File: packages/java/cnos-java-firebase/src/main/java/ai/kitsy/cnos/vault/firebase/FirebaseSecretsProvider.java
package ai.kitsy.cnos.vault.firebase;

import ai.kitsy.cnos.CnosError;
import ai.kitsy.cnos.SecretVaultProvider;
import ai.kitsy.cnos.VaultAuthConfig;
import ai.kitsy.cnos.vault.gcp.GcpSecretManagerProvider;

import java.util.List;
import java.util.Map;

/**
 * CNOS provider for Firebase Secrets.
 * Provider name: {@code firebase-secrets}.
 * Delegates entirely to {@link GcpSecretManagerProvider} — Firebase projects are GCP projects
 * and Firebase Secrets uses the same GCP Secret Manager API.
 */
public final class FirebaseSecretsProvider implements SecretVaultProvider {

    private final SecretVaultProvider delegate;

    FirebaseSecretsProvider(SecretVaultProvider gcpDelegate) {
        this.delegate = gcpDelegate;
    }

    @Override
    public void authenticate(VaultAuthConfig auth) throws CnosError {
        delegate.authenticate(auth);
    }

    @Override
    public Map<String, Object> batchGet(List<String> refs) throws CnosError {
        return delegate.batchGet(refs);
    }

    @Override
    public Object get(String ref) throws CnosError {
        return delegate.get(ref);
    }
}

File: packages/csharp/Kitsy.Cnos.Firebase/FirebaseSecretsProvider.cs
using System;
using System.Collections.Generic;
using Kitsy.Cnos;
using Kitsy.Cnos.Gcp;

namespace Kitsy.Cnos.Firebase
{
    /// <summary>
    /// CNOS vault provider for Firebase Secrets (backed by GCP Secret Manager).
    /// Register via <c>new SecretVaultProviderFactory("firebase", (id, def) => new FirebaseSecretsProvider(projectId))</c>.
    /// </summary>
    public sealed class FirebaseSecretsProvider : ISecretVaultProvider
    {
        private readonly GcpSecretManagerProvider _gcpProvider;

        /// <param name="projectId">Firebase/GCP project ID.</param>
        public FirebaseSecretsProvider(string projectId)
        {
            _gcpProvider = new GcpSecretManagerProvider(projectId);
        }

        public void Authenticate(VaultAuthConfig auth) => _gcpProvider.Authenticate(auth);

        public Dictionary<string, object?> BatchGet(IReadOnlyList<string> refs) => _gcpProvider.BatchGet(refs);

        public object? Get(string reference) => _gcpProvider.Get(reference);

        /// <summary>Returns a <see cref="SecretVaultProviderFactory"/> for this provider.</summary>
        public static SecretVaultProviderFactory Factory(string projectId) =>
            new SecretVaultProviderFactory("firebase", (_, _) => new FirebaseSecretsProvider(projectId));
    }
}
